### ISSUE TITLE
[BUG] In-place transform setters leak gradients through overwritten components

### ISSUE THREAD

**Issue by christophercrouzet** [christophercrouzet/warp#14]
## What

The adjoints for `transform_set_translation()` and `transform_set_rotation()` route the incoming gradient to the replacement value but also leave it on the overwritten component of the original transform. Upstream producers therefore receive gradients for values that no longer affect the output.

## Reproduction

```python
import warp as wp

@wp.kernel
def f(xforms: wp.array[wp.transformf], q: wp.array[wp.quatf], y: wp.array[wp.float32]):
    t = xforms[0]
    wp.transform_set_rotation(t, q[0])
    y[0] = t[3]

for device in ("cpu", "cuda:0"):
    x = wp.array(
        [wp.transformf(wp.vec3f(1,2,3), wp.quatf(4,5,6,7))],
        dtype=wp.transformf, requires_grad=True, device=device,
    )
    q = wp.array([wp.quatf(8,9,10,11)], dtype=wp.quatf, requires_grad=True, device=device)
    y = wp.zeros(1, dtype=wp.float32, requires_grad=True, device=device)
    with wp.Tape() as tape:
        wp.launch(f, dim=1, inputs=[x, q], outputs=[y], device=device)
    tape.backward(y)
    print(device, x.grad.numpy(), q.grad.numpy())
```

Observed original-transform gradient is `[0,0,0,1,0,0,0]` and replacement gradient is `[1,0,0,0]`. The overwritten original quaternion component has no influence on the output and should have derivative zero. Translation exhibits the analogous behavior.

## Why / when it matters

This affects differentiable code that loads a transform, replaces one component in place, then uses the result in a loss. The error propagates into any upstream transform computation.

## Environment

Reproduced against NVIDIA/warp `main` at `b1922eba85bd32af5e393c829c0b22e70ea7abea` (Warp 1.18.0.dev5) on Linux.

Reproduced on CPU and CUDA.
